packages/squiggle-lang/src/js/index.ts

```diff
@@ -1,17 +1,153 @@
+
+// This is here mainly for testing purposes
+export function exportToString(result : tsExport) : string{
+  if(result.tag === "string"){
+    return `"${result.value}"`
+  }
+  else if(result.tag === "boolean"){
+    return `${result.value}`
+  }
+  else if(result.tag === "array"){
+    return `[${result.value.map(exportToString).join(", ")}]`
+  }
+  else if(result.tag === "distribution"){
+    return result.value.toString()
+  }
+  else if(result.tag === "number"){
+    return `${result.value}`
+  }
+  else if(result.tag === "symbol"){
+    return `${result.value}`
+  }
+  else if(result.tag === "record"){
+    return `${_.mapValues(result.value, exportToString)}`
+  }
+}
+
+function expressionValueToValue(value : expressionValue, samplingInputs: SamplingInputs) : tsExport {
+  if(value.tag == "EvArray"){
+    return tagOption("array", value.value.map((val) => expressionValueToValue(val,samplingInputs)));
+  }else if (value.tag == "EvBool"){
+    return tagOption("boolean", value.value)
+  }
+  else if(value.tag == "EvDistribution"){
+    return tagOption("distribution", new Distribution(value.value, samplingInputs));
+  }
+  else if(value.tag == "EvNumber"){
+    return tagOption("number", value.value);
+  }
+  else if(value.tag == "EvString"){
+    return tagOption("string", value.value);
+  }
+  else if(value.tag == "EvSymbol"){
+    return tagOption("symbol", value.value);
+  }
+  else if(value.tag == "EvRecord"){
+    return tagOption("record", _.mapValues(value.value, (val) => expressionValueToValue(val, samplingInputs)));
+  }
+}
+
+export function run(squiggleString : string, samplingInputs : SamplingInputs = defaultSamplingInputs, _environment?: exportEnv): tsExport[] {
+  let result = evaluate(squiggleString);
+  if(result.tag == "Ok"){
+    return [expressionValueToValue(result.value, samplingInputs)];
+  }
+  else{
+    throw Error(errorToString(result.value));
+  }
+}
+
+class Distribution {
+  dist: genericDist
+  samplingInputs : SamplingInputs
+  constructor(dist : genericDist, samplingInputs: SamplingInputs){
+    this.dist = dist;
+    this.samplingInputs = samplingInputs;
+  }
+
+  pointShape() {
+    let pointSet = toPointSet({xyPointLength: this.samplingInputs.outputXYPoints, sampleCount: this.samplingInputs.sampleCount}, this.dist)
+    if(pointSet.tag == "Ok"){
+      return new Shape(pointSet.value);
+    }
+  }
+
+  inv(x: number): number{
+    let result= inv(this.dist, x)
+    if (result.tag == "Ok") {
+      return result.value;
+    }
+    else {
+      throw Error(result.value.toString())
+    }
+  }
+
+  toString(): string {
+    return "Todo"
+  }
+}
+
+type point = { x: number, y: number, cdf: number}
```

Comment:
Having the cdf in this hash is a bit more memory efficient, but I'm not sure its worth the hassle. I think in PointSetDist or DistPlus we already keep a cdf cache. 

Maybe you have some other reason to store this though?